dkron/job.go

```diff
@@ -108,31 +108,37 @@ type Job struct {
 	// Job id of job that this job is dependent upon.
 	ParentJob string `json:"parent_job"`
 
-	// Processors to use for this job
+	// Processors to use for this job.
 	Processors map[string]plugin.Config `json:"processors"`
 
-	// Concurrency policy for this job (allow, forbid)
+	// Concurrency policy for this job (allow, forbid).
 	Concurrency string `json:"concurrency"`
 
-	// Executor plugin to be used in this job
+	// Executor plugin to be used in this job.
 	Executor string `json:"executor"`
 
-	// Executor args
+	// Configuration arguments for the specific executor.
 	ExecutorConfig plugin.ExecutorPluginConfig `json:"executor_config"`
 
-	// Computed job status
+	// Computed job status.
 	Status string `json:"status"`
 
-	// Computed next execution
+	// Computed next execution.
 	Next time.Time `json:"next"`
 
-	ExpiresAt ntime.NullableTime `json:"expires_at"`
+	// Delete the job after the first execution.
+	Ephemeral bool `json:"ephemeral"`
+
+	// The job will not be executed after this time.
+	ExpiresAt time.Time `json:"expires_at"`
```

Comment:
I thought on this, not setting in the API will set its value to the time Zero value, and in the UI we can consider the Zero value as empty and do not show it?

Reply:
From experience with the LastSuccess and LastError fields I know that trying to handle Go's Zero time value in the front end is a real pain to get right. People have different timezones you have to account for in JavaScript - it is just so much easier to handle a null value than to muck about with dates. That's why I eventually chose to solve it with nullable time back then.
Besides that, there should also be something in the UI to prevent users from picking 1-1-1 0:00 as time, because it has special meaning.

As for the API:
What comes out matters. When you don't put a value in, but one (not being null) does come out, that is inconsistent behavior.
What can go in also matters. The API has no UI to help the user, so it should be as self explanatory as possible. If someone puts in 1-1-1 0:00, the sane expectation would be that a job is expired and does not run. But the opposite would be true.

Reply:
absolutely right